pkg/cmd/admin/policy/reconcile_sccs.go

```diff
@@ -243,13 +247,36 @@ func (o *ReconcileSCCOptions) computeUpdatedSCC(expected kapi.SecurityContextCon
 	sort.StringSlice(expected.Groups).Sort()
 	sort.StringSlice(expected.Users).Sort()
 
-	// make a copy of the expected scc here so we can ignore metadata diffs.
+	// compute the updated scc as follows:
+	// 1. start with the expected scc
+	// 2. take the objectmeta from the actual scc (preserves the resource version and uid)
+	// 3. add back the labels and annotations from the expected scc (which were already merged if unioning was desired)
 	updated := expected
-	expected.ObjectMeta = actual.ObjectMeta
+	updated.ObjectMeta = actual.ObjectMeta
+	updated.ObjectMeta.Labels = expected.Labels
+	updated.ObjectMeta.Annotations = expected.Annotations
 
-	if !kapi.Semantic.DeepEqual(expected, actual) {
+	if !kapi.Semantic.DeepEqual(updated, actual) {
 		needsUpdate = true
 	}
 
 	return &updated, needsUpdate
 }
+
+func mergeMaps(a, b map[string]string) map[string]string {
+	if len(a) == 0 && len(b) == 0 {
```

Comment:
only if they're both nil... if you merge a non-nil with a non-nil, you should get a non-nil.